src/sc_stage.py

```diff
@@ -0,0 +1,178 @@
+from threading import Thread, Event
+from queue import Queue, Empty
+from time import sleep
+import logging
+from string import ascii_uppercase, digits
+from random import choice
+from os.path import abspath
+
+from items import Item, ItemDependency
+from file_dict import FileDict
+
+log = logging.getLogger()
+
+class SuppyChainStage(Thread):
+
+    '''
+        SuppyChainStage (SCG) implements the business logic needed
+        by the production stage. Each node maintains a SuppyChainStage that
+        is given a stream of parts/items that get converted into a stream of
+        outputs.
+
+        i.e. SCG mocks a physical production stage that consumes and creates
+        products like a stage in a factory would.
+    '''
+    IN_QUEUE_STAGE = "in-queue"
+    DELIVERED_STAGE = "delivered"
+    CONSUMED_STAGE = "consumed"
+
+    def __init__(self, name, item_dependency: ItemDependency, time_per_unit=1):
+        '''
+            name: debugging purposes only.
+            requirements: the set of items the stage can use as raw material.
+            produces: the item the stage produces.
+            time_per_unit: time in seconds it takes to make one unit.
+            has_inbound:  the stage has incoming raw materials in the supplychain.
+                if not, it mocks an coming stream of raw materials it can
+                consume. e.g. an aluminium plant.
+            has_outbound: the stage has outbound edges. i.e if the resulting product of
+                the stage is consumed by another stage. if not, the inventory
+                count increases indefinitely.
```

Comment:
Thanks for changing that, let's remove these comments as well

Reply:
ack